packages/expo-modules-core/src/requireNativeModule.ts

```diff
@@ -43,3 +43,16 @@ export function requireNativeModule<ModuleType = any>(moduleName: string): Modul
   }
   return nativeModule;
 }
+
+/**
+ * Imports the native module registered with the given name. The same as `requireNativeModule`,
+ * but returns `null` when the module cannot be found instead of throwing an error.
+ *
+ * @param moduleName Name of the requested native module.
+ * @returns Object representing the native module or `null` when it cannot be found.
+ */
+export function requireOptionalNativeModule<ModuleType = any>(
+  moduleName: string
+): ModuleType | null {
+  return globalThis.expo?.modules?.[moduleName] ?? NativeModulesProxy[moduleName] ?? null;
```

Comment:
would be good to rewrite `requireNativeModule` to use this function and throw when the result is null. that would benefit from maintenance. e.g. when we deprecate `NativeModulesProxy`, we don't need to change two places.